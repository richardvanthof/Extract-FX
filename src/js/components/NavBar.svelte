<script>
    import Button from './Button.svelte';

    let {setMode, mode} = $props()

    const modes = ['extract', 'ingest']
</script>

<style lang='scss'>

  .navbar {
    padding: 1em;
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  .nav-links {
    display: flex;
    text-decoration: none;
    list-style: none;
    padding: 0;
    margin: 0;
    border: var(--border-styling);
    border-radius: var(--border-radius-inner);
    overflow: hidden;
  }

  .nav-link {
    background: var(--bg-dark);
    cursor: pointer;
    padding: .66em 1.5em;
    text-decoration: none;

    &:active {
      background: var(--bg-btn-active)
    }

    &.active {
      background: var(--bg-btn-selected);
      border-radius: var(--border-radius);
        &:active {
        background: var(--bg-btn-active)
        }
    }
  }

</style>

<nav class="navbar">
  <ul class="nav-links">
    {#each modes as modeName}
		 <!-- <li class="nav-link class:active={() => mode === 'extract'} onclick={() => setMode('extract')}>Extract</li> -->
      <li class="nav-link {mode === modeName && 'active'}" onclick={()=>setMode(modeName)}>{modeName}</li>
     
    {/each}
    
  </ul>
  <Button title="Help" id="help-btn" onclick="openHelp" name="?" />
</nav>
