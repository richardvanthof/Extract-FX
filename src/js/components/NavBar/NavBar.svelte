<script lang="ts">
    import Button from '../Button.svelte';

    type Props = {
        setMode: (mode: string) => void,
        mode: string,
        options: string[] 
    }

    let {setMode, mode, options=['Extract', 'Ingest']}: Props = $props()

    const openHelp = ():void => {
      console.log('there is no help muhahahahaha')
      //TODO
    }

</script>

<style lang='scss'>

  .navbar {
    padding: 1em;
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  .nav-links {
    display: flex;
    text-decoration: none;
    list-style: none;
    padding: 0;
    margin: 0;
    border: var(--border-styling);
    border-radius: var(--border-radius-inner);
    overflow: hidden;
  }

  .nav-link {
    background: var(--bg-dark);
    cursor: pointer;
    padding: .66em 1.5em;
    text-decoration: none;
    border: var(--border-styling-transparent);
    &:active {
      background: var(--bg-btn-active)
    }

    &.active {
      background: var(--bg-btn-selected);
      border-radius: var(--border-radius);
        &:active {
        background: var(--bg-btn-active)
        }
    }
  }

</style>

<nav class="navbar">
  <ul class="nav-links">
    {#each options as option}  
      <button 
      data-testid="navbar-option" 
      class="nav-link {mode.toLowerCase() === option.toLowerCase() && 'active'}" 
      onclick={()=>setMode(option.toLowerCase())}>
      {option}
    </button>
    {/each}
  </ul>
  <Button title="Help" onclick={openHelp} name="?" />
</nav>
