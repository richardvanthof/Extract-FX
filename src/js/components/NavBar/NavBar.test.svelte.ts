import { render, screen, fireEvent } from '@testing-library/svelte';
import { describe, it, expect, vi } from 'vitest';
import NavBar from './NavBar.svelte';

describe('NavBar', () => {
    
    it('Renders', () => {
        const props = {
            mode: 'extract',
            options: ['Extract', 'Ingest'],
            setMode: vi.fn()
        };
      
        const { container } = render(NavBar, { props });

        const links = screen.getAllByTestId('navbar-option');
        expect(container).toMatchSnapshot();
        expect(links).toHaveLength(2);

    });

    it('Checks if choosing new page works', async () => {
        const props = {
            mode: 'extract',
            options: ['Extract', 'Ingest'],
            setMode: vi.fn((mode: string) => {
                console.log(mode)
                props.mode = mode
            })
        };

        const {rerender} = render(NavBar, {props});

        const links = screen.getAllByTestId('navbar-option');

        expect(links[0].classList).toContain('active');
        expect(links[1].classList).not.toContain('active');

        await fireEvent.click(links[1]);
        await rerender({mode: props.mode})

        expect(links[0].classList).not.toContain('active');
        expect(links[1].classList).toContain('active');
    });
});
