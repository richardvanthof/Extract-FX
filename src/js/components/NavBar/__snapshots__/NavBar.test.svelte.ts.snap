// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar > Renders 1`] = `
<div>
  <nav
    class="navbar svelte-4rlrae"
  >
    <ul
      class="nav-links svelte-4rlrae"
    >
      <button
        class="nav-link active svelte-4rlrae"
        data-testid="navbar-option"
      >
        Extract
      </button>
      <button
        class="nav-link false svelte-4rlrae"
        data-testid="navbar-option"
      >
        Ingest
      </button>
      
    </ul>
     
    <button
      class="svelte-vc9cas"
      name="?"
      title="Help"
    >
      ?
    </button>
    <!---->
  </nav>
  
</div>
`;
